### PR TITLE
Add sequences that have rna16s mentioned in the FASTA header

### PR DESCRIPTION
**Pull request by laughedelic**

### CODE REVIEW

File: src/main/scala/rna16s.scala [hunk 1]
```diff
@@ -46,7 +46,9 @@ case object rna16s extends AnyBlastDB {
   /* Here we want to keep sequences which */
   val predicate: (Row, FASTA.Value) => Boolean = { (row, fasta) =>
      /* - are annotated as encoding 16S */
-     (row.select(gene_name) == geneNameFor16S) &&
+     (row.select(gene_name) == geneNameFor16S ||
+      fasta.getV(header).description.toLowerCase.contains("16s")
```
Comment (eparejatobes) [inline src/main/scala/rna16s.scala]: also add here `small subunit ribosomal RNA`

Comment (laughedelic) [inline src/main/scala/rna16s.scala]: oh.. I just ran it. ok, I'll rerun it tomorrow.

File: src/main/scala/rna16s.scala [hunk 2]
```diff
@@ -47,7 +47,8 @@ case object rna16s extends AnyBlastDB {
   val predicate: (Row, FASTA.Value) => Boolean = { (row, fasta) =>
      /* - are annotated as encoding 16S */
      (row.select(gene_name) == geneNameFor16S ||
-      fasta.getV(header).description.toLowerCase.contains("16s")
+      fasta.getV(header).description.toLowerCase.contains("16s") ||
+      fasta.getV(header).description.toLowerCase.contains("small subunit ribosomal RNA")
```
Comment (laughedelic) [inline src/main/scala/rna16s.scala]: @eparejatobes is it ok?

File: src/main/scala/rna16s.scala [hunk 3]
```diff
@@ -46,7 +46,10 @@ case object rna16s extends AnyBlastDB {
   /* Here we want to keep sequences which */
   val predicate: (Row, FASTA.Value) => Boolean = { (row, fasta) =>
      /* - are annotated as encoding 16S */
-     (row.select(gene_name) == geneNameFor16S) &&
+     (row.select(gene_name) == geneNameFor16S ||
+      fasta.getV(header).description.toLowerCase.contains("16s") ||
+      fasta.getV(header).description.toLowerCase.contains("small subunit ribosomal rna")
```
Comment (laughedelic) [inline src/main/scala/rna16s.scala]: @eparejatobes is it ok?

Comment (eparejatobes) [inline src/main/scala/rna16s.scala]: yes!

Comment (laughedelic) [inline src/main/scala/rna16s.scala]: LOL:

```
$ grep -c "small subunit ribosomal" sources/source.fasta
483603
$ grep -c "small subunit ribosomal RNA" sources/source.fasta
483602
$ grep "small subunit ribosomal [^R]" sources/source.fasta
>URS00001F56CD Plasmodium knowlesi partial small subunit ribosomal rRNA
```

:facepalm:

Comment (eparejatobes) [inline src/main/scala/rna16s.scala]: The joy of bioinformatics

Comment (laughedelic) [inline src/main/scala/rna16s.scala]: Are you sure about this "small subunit" thing? What about these:

```
$ grep "small subunit [^r]" sources/source.fasta  | head
>URS0000004C60 Tropidurus semitaeniatus (striped lava lizard) 12S small subunit RNA
>URS0000006F7E Chlamydomonas reinhardtii small subunit S4
>URS00000110DA Lepidocyrtus paradoxus partial small subunit 12S ribosomal RNA
>URS0000014FAE Uracentron flaviceps 12S small subunit RNA
>URS000006C776 Spiroplasma taiwanense partial small subunit 16S ribosomal RNA
>URS0000098DF9 Pyrenophora teres small subunit 5.8S ribosomal RNA
>URS0000099504 Stenocercus roseiventris 12S small subunit RNA
>URS00000A2B02 Salina tristani partial small subunit 12S ribosomal RNA
>URS00000B6E1F Isotomurus hadriaticus partial small subunit 12S ribosomal RNA
>URS00000C7552 Dicranocentrus marias partial small subunit 12S ribosomal RNA
```

**Comment by laughedelic** [comment]
This is done. The database is at the same place:

```
s3://resources.ohnosequences.com/db/rna16s/
```

It became quite significantly bigger: 253 010 sequences.

```
376M era7bio.db.rna16s.fasta
 11M id2taxa.tsv
blastdb/
   37M era7bio.db.rna16s.fasta.nhr
  2.9M era7bio.db.rna16s.fasta.nin
   89M era7bio.db.rna16s.fasta.nsq
```

File: src/main/scala/rna16s.scala (hunk 3 above)
Comment (eparejatobes) [inline src/main/scala/rna16s.scala]: Ideally we should keep only those with taxonomy ID under bacteria/archaea, but that's too involved (bio4j etc).

It's fine.

**Comment by laughedelic** [comment]
@eparejatobes do you want to check the result? If not, I'm going to merge it tomorrow and use in MG7.

**Comment by eparejatobes** [comment]
LGTM

File: src/main/scala/rna16s.scala (hunk 3 above)
Comment (rtobes) [inline src/main/scala/rna16s.scala]: The small subunit (SSU) of the ribosomal RNA (rRNA) of bacteria and archaea is coded by the 16S gene

```
Type          Size  Large subunit (rRNAs)                                                                 Small subunit (rRNA)
prokaryotic 70S 50S (5S : 120 nt, 23S  : 2906 nt)                                          30S (16S : 1542 nt)
eukaryotic  80S 60S (5S : 121 nt,[1] 5.8S : 156 nt,[2] 28S : 5070 nt[3])           40S (18S : 1869 nt[4])

```

Comment (rtobes) [inline src/main/scala/rna16s.scala]: I don't know the content of the description and RNA type in the table of RNAcentral entries and how of exhaustive is this annotation but in the RNAcentral fasta sequences in the headers of some SILVA and GreenGenes 16S genes the only thing that appears is **rrn** or **rRNA**

Comment (laughedelic) [inline src/main/scala/rna16s.scala]: We add now in this database
- those sequences that have `16S rRNA` annotation in the RNACentral table
- those that have `16s` or `16S` in their header description
- and those that contain `small subunit ribosomal` in their header description

Comment (rtobes) [inline src/main/scala/rna16s.scala]: Then my question is:
The 16S sequences from SILVA and Greengenes that only include in the header of the fasta sequence **rrn** or **rRNA**, are all of them annotated as rRNA in the description column of the table and are selected for our database?

Comment (laughedelic) [inline src/main/scala/rna16s.scala]: I don't know

Comment (rtobes) [inline src/main/scala/rna16s.scala]: > We add now in this database
> those sequences that have 16S rRNA annotation in the RNACentral table
> those that have 16s or 16S in their header description
> and those that contain small subunit ribosomal in their header description

Sorry but I need to known perfectly the composition of the database:
You have added these additional sequences to a primary set composed by what sequences? How the sequences were selected?

Comment (laughedelic) [inline src/main/scala/rna16s.scala]: 1. we take everything from RNACentral
2. filter the table, leaving only those rows that refer to the sequences in the FASTA file (i.e. only those that we can use for Blast)
3. then for this particular 16S database we apply this filter:
   - choosing only 16s:
     - those sequences that have 16S rRNA annotation in the _RNACentral table_
     - **OR** those that have 16s or 16S in their FASTA _header description_ **OR**
     - **OR** and those that contain small subunit ribosomal in their _header description_
   - **AND** the sequence has length > 1000
   - **AND** are not in [this](https://github.com/era7bio/db.rna16S/blob/master/src/main/scala/rna16s.scala#L24-L42) list

**Pull request merged**